src/lexitools/commands/correspondences.py

```diff
@@ -0,0 +1,180 @@
+
+
+def run(args):
+
+    ds = get_dataset(args)
+    clts = args.clts.from_config().api
+    bipa = clts.transcriptionsystem_dict['bipa']
+    color = clts.soundclasses_dict['color']
+    asjp = clts.soundclasses_dict['asjp']
+    sca = clts.soundclasses_dict['sca']
+
+
+    columns=('parameter_id', 'concept_name', 'language_id', 'language_name',
+            'value', 'form', 'segments', 'language_glottocode',
+            'concept_concepticon_id', 'language_latitude',
+            'language_longitude')
+    namespace=(('concept_name', 'concept'), ('language_id', 'doculect'),
+            ('segments', 'tokens'), ('language_glottocode', 'glottolog'),
+            ('concept_concepticon_id', 'concepticon'), ('language_latitude',
+                'latitude'), ('language_longitude', 'longitude'), ('cognacy',
+                    'cognacy'), ('cogid_cognateset_id', 'cogid'))
+
+
+    lex = LexStat.from_cldf(ds.cldf_dir.joinpath('cldf-metadata.json'))
+    args.log.info('Loaded the wordlist')
+
+    G = nx.Graph()
+    for idx, tokens, language in lex.iter_rows('tokens', 'doculect'):
+        for sound in [t.split('/')[1] if '/' in t else t for t in tokens]:
+            try:
+                #G.node[sound]['frequency'] += 1
+                G.node[sound]['language'].add(language)
+            except:
+                G.add_node(
+                        sound, 
+                        frequency=1, 
+                        language=set([language]),
+                        name=bipa[sound].name, 
+                        color=color[sound],
+                        asjp=asjp[sound],
+                        sca=sca[sound],
+                        type=bipa[sound].type
+                        )
+                
+    # add the dummy node for gaps, in case we want to use it
+    G.add_node('-',frequency=0, language=set(lex.cols))
+
+    for node, data in G.nodes(data=True):
+        data['language'] = ', '.join(sorted(data['language']))
+
+
+    for lA, lB in progressbar(combinations(lex.cols, r=2)):
+        for idxA, idxB in lex.pairs[lA, lB]:
+            tokensA, tokensB = (
+                    [t.split('/')[1] if '/' in t else t for t in lex[idxA, 'tokens']],
+                    [t.split('/')[1] if '/' in t else t for t in lex[idxB, 'tokens']]
+                    )
+            langA, langB = lex[idxA, 'doculect'], lex[idxB, 'doculect']
+            classesA, classesB = lex[idxA, 'classes'], lex[idxB, 'classes']
+
+            # check for edit dist == 1
+            if edit_dist(
+                    classesA, 
+                    classesB
+                        ) <= args.threshold:
+                pair = Pairwise(tokensA, tokensB)
+                pair.align()
+                almA, almB, sim = pair.alignments[0]
+                for soundA, soundB in zip(almA, almB):
+                    if soundA != soundB and not '-' in [soundA, soundB] and not '+' in [soundA, soundB]:
+                        G.node[soundA]['frequency'] += 1
```

Comment:
Here we increment frequency not once per occurence of the phoneme in the language, but once per pairwise correspondence. Is it the intent ? 

- If so, the frequency should not be initialized at 1 in the first loop above.
- If the intent was to count occurences of the sound in each language, we need to remove this and instead uncomment the frequency line in the previous loop.

Reply:
Ok, thanks. I'm very glad for your initial code, it solves the interactions I would not have figured out quickly myself (plugging into both Lingpy and lexitools).